Common/src/main/java/me/egg82/tfaplus/services/InternalAPI.java

```diff
@@ -432,46 +446,51 @@ public boolean isVerified(UUID uuid, boolean refresh) {
         return Optional.of(Boolean.FALSE);
     }
 
-    private void setHOTP(UUID uuid, long length, long counter, SecretKey key, JedisPool redisPool, ConfigurationNode redisConfigNode, Connection rabbitConnection, SQL sql, ConfigurationNode storageConfigNode, SQLType sqlType, boolean debug) {
-        if (debug) {
+    private void setHOTP(UUID uuid, long length, long counter, SecretKey key) {
+        if (ConfigUtil.isDebugging()) {
             logger.info("Setting new HOTP counter for " + uuid);
         }
 
         // SQL
         HOTPData result = null;
-        try {
-            if (sqlType == SQLType.MySQL) {
-                result = MySQL.updateHOTP(sql, storageConfigNode, uuid, length, counter, key).get();
-            } else if (sqlType == SQLType.SQLite) {
-                result = SQLite.updateHOTP(sql, storageConfigNode, uuid, length, counter, key).get();
-            }
-        } catch (ExecutionException ex) {
-            logger.error(ex.getMessage(), ex);
-        } catch (InterruptedException ex) {
-            logger.error(ex.getMessage(), ex);
-            Thread.currentThread().interrupt();
-        }
+        result = getHotpData(uuid, length, counter, key, result);
 
         if (result == null) {
             return;
         }
 
         // Redis
-        Redis.update(result, redisPool, redisConfigNode);
+        Redis.update(result, ConfigUtil.getRedisPool(), ConfigUtil.getRedisConfigNode());
 
         // Rabbit
-        RabbitMQ.broadcast(result, rabbitConnection);
+        broadcastResult(result);
```

Comment:
Redis & Rabbit broadcast/update should probably be in one method, if we're going to make it a method anyway.